python/pyspark/sql/udf.py

```diff
@@ -405,12 +409,10 @@ def register(
                 PythonEvalType.SQL_SCALAR_PANDAS_UDF,
                 PythonEvalType.SQL_SCALAR_PANDAS_ITER_UDF,
                 PythonEvalType.SQL_GROUPED_AGG_PANDAS_UDF,
-                PythonEvalType.SQL_MAP_PANDAS_ITER_UDF,
             ]:
                 raise ValueError(
                     "Invalid f: f must be SQL_BATCHED_UDF, SQL_SCALAR_PANDAS_UDF, "
-                    "SQL_SCALAR_PANDAS_ITER_UDF, SQL_GROUPED_AGG_PANDAS_UDF or "
-                    "SQL_MAP_PANDAS_ITER_UDF."
```

Comment:
This change seems unrelated and a regression, did PANDAS_ITER_UDF not work here?

Reply:
Oh I had to clarify this. Yeah `SQL_MAP_PANDAS_ITER_UDF` is only used for `DataFrame.mapInPandas` which cannot be registered as a UDF. I will clarify this in Pr description.